lib/client-options.js

```diff
@@ -179,6 +179,26 @@ function createQueryOptions(client, userOptions, rowCallback, logged) {
     // not part of query options
     rowCallback: rowCallback
   };
+  if (userOptions === utils.emptyObject) {
+    return result;
+  }
+  var userOptionsKeys = Object.keys(userOptions);
+  var key, value;
+  // Use the fastest iteration of array
+  var i = userOptionsKeys.length;
+  while (i--) {
+    key = userOptionsKeys[i];
+    if (key === 'executionProfile') {
+      // Execution profile was the only value that could has been "replaced"
+      continue;
+    }
+    value = userOptions[key];
+    if (value === undefined) {
+      continue;
+    }
+    result[key] = value;
```

Comment:
It seems safe to override `result[key]` if it exists because the default behavior is to use userOptions anyways (in all cases but executionProfile like you documented above) 👍